manifests/config.pp

```diff
@@ -23,13 +16,30 @@
     },
   }
 
-  file { 'otelcol-config' :
-    ensure  => 'file',
-    path    => $otelcol::config_file,
-    content => template('otelcol/config.yml.erb'),
-    owner   => $otelcol::config_file_owner,
-    group   => $otelcol::config_file_group,
-    mode    => $otelcol::config_file_mode,
+  concat { 'otelcol-config' :
+    ensure => present,
+    path   => $otelcol::config_file,
+    format => 'yaml',
+    owner  => $otelcol::config_file_owner,
+    group  => $otelcol::config_file_group,
+    mode   => $otelcol::config_file_mode,
+  }
+  concat::fragment { 'otelcol-config-header' :
+    target  => 'otelcol-config',
+    order   => 0,
+    content => template('otelcol/config-header.yml.erb'),
+  }
+
+  concat::fragment { 'otelcol-config-baseconfig' :
+    target  => 'otelcol-config',
+    order   => 10000,
+    content => stdlib::to_yaml($component),
+  }
+
+  concat::fragment { 'otelcol-config-footer' :
+    target  => 'otelcol-config',
+    order   => 10001,
+    content => template('otelcol/config-footer.yml.erb'),
```

Comment:
Same here:
```suggestion
    content => '',
```

Though I wonder what the benefit of this is. Did you intend to use it to ensure a newline?

Reply:
Removed Footer at all, makes no sense. TY